st4_py38/lsp_utils/server_npm_resource.py

```diff
@@ -99,6 +97,11 @@ def needs_installation(self) -> bool:
                 raise Exception('Missing required "package.json" in {}'.format(self._server_src))
             src_hash = md5(src_package_json.read_bytes()).hexdigest()
             try:
+                with open(self._node_version_marker_file) as file:
+                    node_version = str(self._node_runtime.resolve_version())
+                    stored_node_version = file.read()
+                    if node_version != stored_node_version.strip():
+                        return True
```

Comment:
the old code read in this order:
1. read file in Packages folder
1. read file in Package Storage folder
1. compare the hash

the new code read in this order:
1. read file in Packages folder
1. **(new)** did the node version change, if so return.
1. read file in Package Storage folder
1. compare the hash

Do you thing that moving lines 100-104, 
before the  `# Server already installed` comment would make the code a slight bit better?
1. **(new)** did the node version change, if so return.
1. read file in Packages folder
1. read file in Package Storage folder
1. compare the hash

Reply:
I'm not sure if moving the code before `# Server already installed` would be better but I've moved the `src_hash` line down, closer to where it's used. If moving before the comment line then we'd need to duplicate the try/except block.